Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm

```diff
@@ -1470,8 +1470,12 @@ static RecommendDesktopClassBrowsingForRequest desktopClassBrowsingRecommendedFo
     if (Quirks::needsIPadMiniUserAgent(request.url()))
         return RecommendDesktopClassBrowsingForRequest::No;
 
-    if (Quirks::needsIPhoneUserAgent(request.url()))
-        return RecommendDesktopClassBrowsingForRequest::No;
+    if (Quirks::needsIPhoneUserAgent(request.url())) {
+        const API::WebsitePolicies& policies = m_configuration->defaultWebsitePolicies();
+        policies.setCustomUserAgent(makeStringByReplacingAll(standardUserAgentWithApplicationName(m_applicationNameForUserAgent), "iPad"_s, "iPhone"_s));
+        policies.setCustomNavigatorPlatform("iPhone"_s);
+        return WebContentMode::Mobile;
```

Comment:
Why are we returning a WebContentMode here?

The function returns a RecommendDesktopClassBrowsingForRequest.

Reply:
Yes: You should change this to:
`return RecommendDesktopClassBrowsingForRequest::No;` like it was before you change. That should give you the behavior you want.